cranelift/codegen/src/machinst/buffer.rs

```diff
@@ -236,6 +238,7 @@ impl<I: VCodeInst> MachBuffer<I> {
     pub fn put1(&mut self, value: u8) {
         trace!("MachBuffer: put byte @ {}: {:x}", self.cur_offset(), value);
         self.data.push(value);
+        self.labels_at_tail.clear();
```

Comment:
Seems like significant added expense on a hot path -- avoidable?

Reply:
Good point! Now using a lazy-clear scheme: we store the offset at which the label vector applies; if `cur_offset()` moves, the vector is conceptually empty, and is made so next time we need to read it.